internal/identify/identify.go

```diff
@@ -62,64 +67,103 @@ type scrapeResult struct {
+func (t *SceneIdentifier) getOptions(source *ScraperSource) MetadataOptions {
+	options := t.DefaultOptions
+	if source.Options == nil {
+		return *options
+	}
+	if source.Options.SetCoverImage != nil {
+		options.SetCoverImage = source.Options.SetCoverImage
+	}
+	if source.Options.SetOrganized != nil {
+		options.SetOrganized = source.Options.SetOrganized
+	}
+	if source.Options.IncludeMalePerformers != nil {
+		options.IncludeMalePerformers = source.Options.IncludeMalePerformers
+	}
+	if source.Options.IncludeMalePerformers != nil {
+		options.IncludeMalePerformers = source.Options.IncludeMalePerformers
+	}
+	if source.Options.SkipMultipleMatches != nil {
+		options.SkipMultipleMatches = source.Options.SkipMultipleMatches
+	}
+	if source.Options.SkipMultipleMatchTag != nil && len(*source.Options.SkipMultipleMatchTag) > 0 {
+		options.SkipMultipleMatchTag = source.Options.SkipMultipleMatchTag
+	}
+	if source.Options.SkipSingleNamePerformers != nil {
+		options.SkipSingleNamePerformers = source.Options.SkipSingleNamePerformers
+	}
+	if source.Options.SkipSingleNamePerformerTag != nil && len(*source.Options.SkipSingleNamePerformerTag) > 0 {
+		options.SkipSingleNamePerformerTag = source.Options.SkipSingleNamePerformerTag
+	}
+	return *options
+}
+
 func (t *SceneIdentifier) getSceneUpdater(ctx context.Context, s *models.Scene, result *scrapeResult) (*scene.UpdateSet, error) {
 	ret := &scene.UpdateSet{
 		ID: s.ID,
 	}
 
-	options := []MetadataOptions{}
+	allOptions := []MetadataOptions{}
 	if result.source.Options != nil {
-		options = append(options, *result.source.Options)
+		allOptions = append(allOptions, *result.source.Options)
 	}
 	if t.DefaultOptions != nil {
-		options = append(options, *t.DefaultOptions)
+		allOptions = append(allOptions, *t.DefaultOptions)
 	}
 
-	fieldOptions := getFieldOptions(options)
-
-	setOrganized := false
-	for _, o := range options {
-		if o.SetOrganized != nil {
-			setOrganized = *o.SetOrganized
-			break
-		}
-	}
+	fieldOptions := getFieldOptions(allOptions)
+	options := t.getOptions(&result.source)
 
 	scraped := result.result
 
 	rel := sceneRelationships{
-		sceneReader:      t.SceneReaderUpdater,
-		studioCreator:    t.StudioCreator,
-		performerCreator: t.PerformerCreator,
-		tagCreator:       t.TagCreator,
-		scene:            s,
-		result:           result,
-		fieldOptions:     fieldOptions,
+		sceneReader:              t.SceneReaderUpdater,
+		studioCreator:            t.StudioCreator,
+		performerCreator:         t.PerformerCreator,
+		tagCreator:               t.TagCreator,
+		scene:                    s,
+		result:                   result,
+		fieldOptions:             fieldOptions,
+		skipSingleNamePerformers: *options.SkipSingleNamePerformers,
 	}
 
-	ret.Partial = getScenePartial(s, scraped, fieldOptions, setOrganized)
+	ret.Partial = getScenePartial(s, scraped, fieldOptions, *options.SetOrganized)
```

Comment:
Isn't `options.SetOrganized` potentially `nil` here? `getOptions` overrides the `DefaultOptions` if they are not `nil`, but as far as I can tell, it could still be `nil` if neither the source options or the default options have this field set to a non-nil value.